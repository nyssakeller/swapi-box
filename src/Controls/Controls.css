@import url('https://fonts.googleapis.com/css?family=Orbitron');

button {
  border-radius: 10px;
  background-color: #FFA1A1;
  border-color: #FFA1A1;
  box-shadow: 0px 0px 50px #FD0101;
  color: white;
  font-family: Orbitron;
  font-size: 15px;
  height: 40px;
  margin: 30px 20px;
  width: 120px;
}

button:hover {
  background-color: #9AEAFF;
  border-color: #9AEAFF;
  box-shadow: 0px 0px 50px #013EFE;

}

.active {
  background-color: #9AEAFF;
  border-color: #9AEAFF;
  box-shadow: 0px 0px 50px #013EFE;
}