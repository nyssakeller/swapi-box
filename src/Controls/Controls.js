import React from 'react';
import './Controls.css';

const Controls = ({getButtonClass}) => {
  return (
    <div>
      <button className='people' onClick={(e) => getButtonClass(e.target.className, e.target)}>people</button>
      <button className='planets' onClick={(e) => getButtonClass(e.target.className, e.target)}>planets</button>
      <button className='vehicles' onClick={(e) => getButtonClass(e.target.className, e.target)}>vehicles</button>
      <button className='favorite' onClick={(e) => getButtonClass(e.target.className, e.target)}>favorites</button>
    </div>
  )
}

export default Controls;