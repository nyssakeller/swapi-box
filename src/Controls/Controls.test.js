/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls.js';

describe('Controls', () => {

  const mockFavorites = [{}, {}, {}];
  
  it('snapshot test', () => {
    const wrapper = shallow(<Controls favorites={mockFavorites}/>);
    expect(wrapper).toMatchSnapshot();
  });

});

