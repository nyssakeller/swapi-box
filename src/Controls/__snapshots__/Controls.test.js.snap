// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    favorites={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="people"
          onClick={[Function]}
>
          people
</button>,
        <button
          className="planets"
          onClick={[Function]}
>
          planets
</button>,
        <button
          className="vehicles"
          onClick={[Function]}
>
          vehicles
</button>,
        <button
          className="favorites"
          onClick={[Function]}
>
          favorites: 
          3
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
          "className": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "people",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "planets",
          "className": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "vehicles",
          "className": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "favorites: ",
            3,
          ],
          "className": "favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "favorites: ",
          3,
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people"
            onClick={[Function]}
>
            people
</button>,
          <button
            className="planets"
            onClick={[Function]}
>
            planets
</button>,
          <button
            className="vehicles"
            onClick={[Function]}
>
            vehicles
</button>,
          <button
            className="favorites"
            onClick={[Function]}
>
            favorites: 
            3
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
            "className": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "people",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
            "className": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "vehicles",
            "className": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "favorites: ",
              3,
            ],
            "className": "favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "favorites: ",
            3,
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
